docs/advanced-topics/advanced-methods.md

```diff
@@ -0,0 +1,93 @@
+
+Lets first take a look at our default loss function, Triplet Margin Loss.  
+
+Triplet Margin Loss is a *contrastive* loss function, meaning that the loss is calculated by comparing the embeddings of multiple documents, three, to be exact.
+Each triplet of documents consists of an anchor document, a positive document and a negative document.
+The anchor and the positive document belong to the same class, and the negative document belongs to a different class.
+The goal of Triplet margin loss is to maximise the difference between the distance from the anchor to the positive document, and the distance from the anchor to the negative document.
+For a more detailed explanation on Triplet Loss, as well as how samples are gathered to create these Triplets, see {doc}`/advanced_topics/negative-mining`.
+
+### SphereFace losses
+
+The other two loss functions that we support, ArcFace and CosFace, are both variations on the SphereFace loss function.
+Instead of attempting to minimise the distance between positive pairs and maximise the distance between negative pairs, the SphereFace loss functions compare each sample with the center of each class
+and attempts to minimize the *angular distance* between the document and its class centroid, and maximise the angular distance between the document and the centroids of the other classes.
+
+![training](../imgs/SphereFace-training.png)
+
+The ArcFace and CosFace both deviate from the traditional SphereFace by including a margin and scaling parameter, which can be used to increase the boundary between each class. If an item's embedding is within the boundary of the class it belongs to, then no loss is incurred. Choosing appropriate values for the margin and scaling parameter is important for effective training, for more information on how ArcFace and CosFace calculate loss, and how these parameters affect the output, see this article on [loss metrics for deep learning](https://hav4ik.github.io/articles/deep-metric-learning-survey#cosface).  
+
+Since only one sample is needed to calculate the loss with the ArcFace and CosFace functions, there are no constraints on what each batch needs to contain, unlike Triplet Margin Loss, which uses a `ClassSampler` to construct batches an equal amount of each class in the batch.
+Instead, we can construct batches using random sampling, a much simpler method which consequently takes less time to construct a batch.
+By default, runs created ArcFace or Cosface loss will use random sampling, however you can specify which type of sampling method you would like to use like so:
+
+```diff
+run = finetuner.fit(
+    ...,
+    loss='ArcFaceLoss',
++   sampler = 'random'     # use random sampling
++   sampler = 'class'      # use class sampling
++   sampler = 'random'     # infer sampling method based on the loss function (default)
+
+)
+```
+
+Since Triplet Margin Loss needs a class sampler to properly function, this `sampler` parameter is ignored when the `loss=TripletMarginLoss`.
```

Comment:
```suggestion
Since Triplet Margin Loss needs a class sampler to properly function, this `sampler` parameter is ignored when `loss=TripletMarginLoss`.
```